### ISSUE TITLE
`dispatcher.utter_message` in a slot validation action does not display in a conversation

### ISSUE THREAD

**Issue by ancalita** [RasaHQ/rasa-sdk#612]
**Problem description**:
Using a `dispatcher.utter_message(text=..)` inside a custom validation method in a custom action extending `ValidationAction` does not get displayed to the user in a `rasa shell` conversation, although the debug log shows that the `BotUttered` event was added to the tracker.

Example of custom action code:
```
class TravelSlotValidation(ValidationAction):
    @staticmethod
    def location_db() -> List[Text]:
        """Database of supported locations."""
        return ["copenhagen", "vancouver", "london", "paris", "lima",
                "san sebastian", "aspen", "queenstown", "new york", "milan"]

    def validate_location(
        self,
        slot_value: Any,
        dispatcher: CollectingDispatcher,
        tracker: Tracker,
        domain: DomainDict,
    ) -> Dict[Text, Any]:
        if slot_value.lower() in self.location_db():
            return {"location": slot_value}
        else:
            dispatcher.utter_message(text='Your city is not yet supported. Please try again')
            return {"location": None}
```

**Definition of Done**
- [ ] bot message is shown to user in `rasa shell`.

**Comment by rajpar29** [comment]
Do we have any updates on this?

**Comment by ancalita** [comment]
Fix released in rasa-oss version `3.2.5`.

**Comment by tedd0n** [comment]
Still having the same problem. Rasa version 3.6.13
